messaging/kafka-producer/src/test/java/io/quarkus/ts/messaging/kafka/producer/SnappyCompressionIT.java

```diff
@@ -0,0 +1,163 @@
+        GenericContainer<?> container = kafka.getPropertyFromContext(DockerContainerManagedResource.DOCKER_INNER_CONTAINER);
+        int originalJsonSize = calculateOriginalJsonLength();
+        String largeJson = loadLargeJsonFromFile(BIG_JSON);
+        UniAssertSubscriber<Object> subscriber = makeHttpReqWithMessage("/messageEvent", largeJson)
+                .subscribe().withSubscriber(UniAssertSubscriber.create());
+        subscriber.awaitItem(Duration.ofSeconds(TIMEOUT_SEC)).getItem();
+
+        String logSegmentHead = container.execInContainer(COMMAND_LOG_CONTAINER)
+                .getStdout();
+        int compressedMessageSize = extractCompressedSize(logSegmentHead);
+        Assertions.assertTrue(compressedMessageSize < originalJsonSize,
+                "Compressed message size should be smaller than the original size");
+    }
+
+    private KafkaConsumer<Integer, String> createConsumer() {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapUrl());
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, "test");
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        KafkaConsumer<Integer, String> consumer = new org.apache.kafka.clients.consumer.KafkaConsumer<>(props);
+        consumer.subscribe(Collections.singletonList("test"));
+        return consumer;
+    }
+
+    private int extractCompressedSize(String logSegmentHead) throws IOException {
+
+        Pattern pattern = Pattern.compile("size: (\\d+)");
+        Matcher matcher = pattern.matcher(logSegmentHead);
+
+        if (matcher.find()) {
+            return Integer.parseInt(matcher.group(1));
+        } else {
+            throw new IOException("Failed to extract compressed size from log segment head: " + logSegmentHead);
+        }
+    }
+
+    private String loadLargeJsonFromFile(String filePath) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(filePath)) {
+            if (inputStream != null) {
+                return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            } else {
+                throw new FileNotFoundException("File not found: " + filePath);
+            }
+        }
+    }
+
+    private int calculateOriginalJsonLength() throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(BIG_JSON)) {
+            if (inputStream == null) {
+                throw new IOException("Resource not found: " + BIG_JSON);
+            }
+            byte[] jsonBytes = inputStream.readAllBytes();
+            System.out.println("BIG " + jsonBytes.length);
```

Comment:
System.out definitely can't be here. Please remove this leftover. If you need to print something use logger.